Month1/Week2/Section2/Month1-Week2-Section2/Spotify/Spotify.cpp

```diff
@@ -0,0 +1,241 @@
+        std::cout << "Please enter the song name: " << std::endl;
+        std::getline(std::cin, EntrySongName);
+        while (!std::cin.good() || EntrySongName == "" || std::cin.peek() == '\n')
+        {
+            std::cout << "Invalid Input or Empty space, please enter the song name again: " << std::endl;
+            std::cin.clear();
+            std::cin.ignore(1000, '\n');
+            std::getline(std::cin, EntrySongName);
+        }
+        std::cout << "Please enter the artist name: " << std::endl;
+        std::getline(std::cin, EntryArtistName);
+        while (EntryArtistName == "" || !std::cin.good() || std::cin.peek() == '\n')
+        {
+            std::cout << "Invalid Input or Empty space, please enter the artist name again: " << std::endl;
+            std::cin.clear();
+            std::cin.ignore(1000, '\n');
+            std::getline(std::cin, EntryArtistName);
+        }
+        std::cout << "Please enter the song duration in minutes and seconds" << std::endl;
+        std::cout << "Minutes: " << std::endl;
+        std::cin >> EntryMinutesDuration;
+        while (EntryMinutesDuration < 0 || !std::cin.good() )
+        {
+            std::cout << "Invalid input, remember the duration must be greater of cero  " << std::endl;
+            std::cin.clear();
+            std::cin.ignore(1000, '\n');
+            std::cin >> EntryMinutesDuration;
+        }
+        std::cout << "Seconds: " << std::endl;
+        std::cin >> EntrySecondsDuration;
+        while (!std::cin.good() || EntrySecondsDuration > 60 || EntrySecondsDuration < 0 )
+        {
+            std::cout << "Invalid input, remember the seconds duration must be lower of 60 and equal o greater of cero " << std::endl;
+            std::cout << "Please enter Your second sduration again:" << std::endl;
+            std::cin.clear();
+            std::cin.ignore(1000, '\n');
+            std::cin >> EntrySecondsDuration;
+        }
+        while (EntryMinutesDuration == 0 && EntrySecondsDuration == 0)
+        {
+            std::cout << "Remember te duration of the song can not be cero. Enter Your Song Duration again (minutes and seconds ): " << std::endl;
+            std::cout << "Minutes: " << std::endl;
+            std::cin >> EntryMinutesDuration;
+            std::cout << "Seconds:" << std::endl;
+            std::cin >> EntrySecondsDuration;
+        }
+        FSong NewSong(EntryMinutesDuration, EntrySecondsDuration, EntryArtistName, EntrySongName);
+        ANewPlayList.bAddSong(NewSong);
+        std::cout << "Song add sucefull, What do you whant to do now?" << std::endl;
+        std::cout << "1. Enter another song" << std::endl;
+        std::cout << "2.Go to main menu" << std::endl;
+        std::cin >> AddSongs;
+      
+        std::cin.clear();
+        std::cin.ignore(1000, '\n');
+        system("CLS");
+    } while (AddSongs == 1);
+    PlayListContainer.AddPlaylist(ANewPlayList);
+    ActiveCommand = MenuCommand::MainMenu;
+}
```

Comment:
This method is very nice! But I would suggest using a bit more spaces. It is hard to follow the function since there are hardly any spaces, so you get no breaks when reading through it.

Reply:
Spaces Added